src/coreclr/src/jit/codegencommon.cpp

```diff
@@ -1172,19 +1172,7 @@ unsigned CodeGenInterface::InferStructOpSizeAlign(GenTree* op, unsigned* alignme
             }
             else
             {
-                opSize       = (unsigned)op2->AsIntCon()->gtIconVal;
-                GenTree* op1 = op->AsOp()->gtOp1;
-                assert(op1->OperGet() == GT_LIST);
-                GenTree* dstAddr = op1->AsOp()->gtOp1;
-                if (dstAddr->OperGet() == GT_ADDR)
-                {
-                    InferStructOpSizeAlign(dstAddr->AsOp()->gtOp1, &alignment);
-                }
-                else
-                {
-                    assert(!"Unhandle dstAddr node");
-                    alignment = TARGET_POINTER_SIZE;
-                }
+                unreached();
```

Comment:
Yep, this is one of the things I need to investigate a bit more. I suspect this is a leftover from GT_COPYBLOCK days.

Reply:
Yup, copy block used to have a list as the first operand:
```
    /* Note  that this use of a  GT_LIST is different than all others */
    /* in that the the GT_LIST is used as a tuple [dest,src] rather   */
    /* than a being a NULL terminated list of GT_LIST nodes           */
    result->gtOp.gtOp1 = gtNewOperNode(GT_LIST, TYP_VOID,  /*        GT_[oper]          */
                                       dst, srcOrFillVal); /*        /      \           */
    result->gtOp.gtOp2 = hndOrSize;                        /*   GT_LIST      \          */
```
Not only that but it used `LIST` in an unusual manner, that explains some weird `LIST` code I encountered. Good riddance.

Even if this function is called with a `CopyBlockOp` node this code path still can't be hit because a copy block with a constant `gtOp2` is actually `InitBlkOp`. We would just hit `noway_assert(!"Variable sized COPYBLK register arg!");` below.

Reply:
> Not only that but it used LIST in an unusual manner, that explains some weird LIST code I encountered. Good riddance.

Yes, that was one of the early struct-related changes, and trying to do it in an incremental fashion, initially as zero-diff, is also responsible for some of the weird `TODO-1stClassStructs` remnants.

Reply:
> that was one of the early struct-related changes

Hmm, not sure I understand. Are you saying that that particular use of `LIST` was the result of early struct related changes and there was yet another way to represent copy block before?

Can't say I've seen any weird TODO, I was thinking of some code (e.g. `gtSetListOrder`) that appears to try to handle a `LIST` node where `gtOp2` isn't a `LIST` as well.

Not that it would matter today, just curious how it all evolved :)

Reply:
No - struct copies used to be `GT_COPYBLK(GT_LIST(dstAddr,srcAddr),size)`, which is the case you mention where op1 isn't a `LIST`. This also resulted in odd order options for `GT_DYN_BLK` (ABC, BAC, CAB or CBA but not ACB or BCA if A is dstAddr, B is srcAddr and C is size) which resulted in adding `gtEvalSizeFirst` for `GenTreeDynBlk` to duplicate previous behavior and which probably didn't even matter since they're so rare. Maybe there was never a TODO associated with that.